packages/client/hmi-client/src/utils/number.ts

```diff
@@ -66,5 +70,24 @@ export function displayNumber(num: string): string {
 	digitString = digitString.replace('.', '');
 
 	if (digitString.length > 6) return parseFloat(num).toExponential(3);
-	return numberToNist(num);
+	return numberToNist(parseFloat(num).toString());
+}
+
+export function scrubAndParse(v) {
+	const scrubbed = v.replace(/\s/g, '');
+
+	// check if last digit is a decimal point
+	if (scrubbed[scrubbed.length - 1] === '.') return false;
```

Comment:
Why is last character being `.` a problem? Isn't `123.` or `3.` just a floating point number?

Reply:
I mean NIST is not clear on the subject [paragraph 10.5.2 Decimal sign or marker](https://www.nist.gov/pml/special-publication-811/nist-guide-si-chapter-10-more-printing-and-using-symbols-and-numbers)

I do agree that we should not parse the input until the full number is entered. A user would not enter a `.` without any value behind. Or if it does, the on blur will make it a number

Reply:
from the code if the user were to type in `123.` this would evaluate as true and then call the numberToNist function which will filter out the decimal point back to `123`.